src/main/java/uk/gov/companieshouse/config/SecurityConfig.java

```diff
@@ -4,30 +4,39 @@
 import org.springframework.context.annotation.Configuration;
 import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
 import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+import uk.gov.companieshouse.api.interceptor.InternalUserInterceptor;
 import uk.gov.companieshouse.api.interceptor.TokenPermissionsInterceptor;
 import uk.gov.companieshouse.interceptor.DissolutionTokenPermissionsInterceptor;
 
 @Configuration
 public class SecurityConfig implements WebMvcConfigurer {
 
-    private static final String URI_PATTERN = "/dissolution-request/*";
+    private static final String URI_PATTERN = "/dissolution-request/**";
 
-    private static final String[] TOKEN_PERMISSION_AUTH_WHITELIST = {
+    private static final String[] TOKEN_PERMISSION_AUTH_EXCLUDE_LIST = {
             "/dissolution-request/healthcheck",
-            "/dissolution-request/{company-number}/payment",
+            "/dissolution-request/{company-number:[0-9]+}/payment",
             "/dissolution-request/submit"
     };
 
+    private static final String[] API_KEY_PERMISSION_AUTH_EXCLUDE_LIST = {
+            "/dissolution-request/healthcheck",
+            "/dissolution-request/{company-number:[0-9]+}"
```

Comment:
Previously, `/dissolution-request/{company-number}` was actually excluding `/dissolution-request/submit` from the API Key Interceptor as well (they were both being treated as the same URI pattern).
So needed to update our `company-number` path variable to say this data type will be a number.
More info on this here, https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/web/util/pattern/PathPattern.html

Reply:
This isn't true, LLDS01 company numbers start with LL